spot-client/src/common/utils/store-peristence.test.js

```diff
@@ -0,0 +1,135 @@
+import { persistence } from 'common/utils';
+import { getPersistedState } from './store-persistence';
+
+describe('store-persistence', () => {
+    const SPOT_TV_PERM_CODE = '123456';
+    const SPOT_REMOTE_PERM_CODE = '654321';
+    const COMMON_BACKEND_PERM_CODE = '12345678';
+
+    describe('restores legacy pairing code', () => {
+        it('when both remote and tv keys are present', () => {
+            jest.spyOn(persistence, 'get')
+                .mockReturnValue({
+                    'spot-tv/backend': {
+                        permanentPairingCode: SPOT_TV_PERM_CODE
+                    },
+                    spotRemote: {
+                        permanentPairingCode: SPOT_REMOTE_PERM_CODE
+                    }
+                });
+
+            expect(getPersistedState()).toMatchObject({
+                backend: {
+                    permanentPairingCode: SPOT_TV_PERM_CODE
+                }
+            });
+        });
```

Comment:
Nice. Maybe we want a test that fails in a few weeks as a reminder to delete the legacy support.